test/test_transformer.py

```diff
@@ -66,6 +68,49 @@ def get_layers(module):
     return layers
 
 
+def pipe_to_sd(pipe):
+    sd = {}
+    for stage_idx in range(pipe.num_stages):
+        stage_mod = pipe.get_stage_module(stage_idx)
+        sd[f"stage_{stage_idx}"] = stage_mod
```

Comment:
not really clear to me why we need to add a prefix at all.

orig model
Transformer
  embedding
  layers
    0 
    1

Split model

  submod0
     embedding
     layers
        0

  submod 1
    layers
      1

There should be no duplication of fqns between submods/stages.

what are we doing about the 'submod_0' part in the fqn?  when we do `stage_mod = pipe.get_stage_module(stage_idx)` does that return us a module that has top level keys like `embedding` and `layers` or a module that has a top level key of `submod_n`?  

If the former, can't we just save/load the keys as usual?  

If the latter, we can still save/load without a prefix of `stage_{idx}` i think, but we'll sadly be uncompatible to load into a non-PP model later on if we want to.

Reply:
Former. @wconstab